src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -447,8 +448,9 @@ private void logDuration(List<Course> courses, User user, long timeNanoStart) {
             Map<ExerciseMode, List<Exercise>> exercisesGroupedByExerciseMode = exercises.stream().collect(Collectors.groupingBy(Exercise::getMode));
             int noOfIndividualExercises = Objects.requireNonNullElse(exercisesGroupedByExerciseMode.get(ExerciseMode.INDIVIDUAL), List.of()).size();
             int noOfTeamExercises = Objects.requireNonNullElse(exercisesGroupedByExerciseMode.get(ExerciseMode.TEAM), List.of()).size();
-            log.info("/courses/for-dashboard finished in {} for {} courses with {} individual exercises and {} team exercises for user {}",
-                    TimeLogUtil.formatDurationFrom(timeNanoStart), courses.size(), noOfIndividualExercises, noOfTeamExercises, user.getLogin());
+            Set<Exam> exams = courses.stream().flatMap(course -> course.getExams().stream()).collect(Collectors.toSet());
+            log.info("/courses/for-dashboard finished in {} for {} courses with {} individual exercise(s), {} team exercise(s), and {} exam(s) for user {}",
+                    TimeLogUtil.formatDurationFrom(timeNanoStart), courses.size(), noOfIndividualExercises, noOfTeamExercises, exams.size(), user.getLogin());
```

Comment:
What about this? Would avoid an useless intermediary ``Set`` and ``n`` intermediary streams
```suggestion
            int examCount = courses.stream().mapToInt(course -> course.getExams().size()).sum();
            log.info("/courses/for-dashboard finished in {} for {} courses with {} individual exercise(s), {} team exercise(s), and {} exam(s) for user {}",
                    TimeLogUtil.formatDurationFrom(timeNanoStart), courses.size(), noOfIndividualExercises, noOfTeamExercises, examCount, user.getLogin());
```